homeassistant/components/openai_conversation/conversation.py

```diff
@@ -187,23 +199,61 @@ async def _async_handle_message(
             ]
 
         messages = [
-            _chat_message_convert(message) for message in session.async_get_messages()
+            _chat_message_convert(message, model)
+            for message in session.async_get_messages()
         ]
 
         client = self.entry.runtime_data
 
         # To prevent infinite loops, we limit the number of iterations
         for _iteration in range(MAX_TOOL_ITERATIONS):
             try:
-                result = await client.chat.completions.create(
-                    model=options.get(CONF_CHAT_MODEL, RECOMMENDED_CHAT_MODEL),
-                    messages=messages,
-                    tools=tools or NOT_GIVEN,
-                    max_tokens=options.get(CONF_MAX_TOKENS, RECOMMENDED_MAX_TOKENS),
-                    top_p=options.get(CONF_TOP_P, RECOMMENDED_TOP_P),
-                    temperature=options.get(CONF_TEMPERATURE, RECOMMENDED_TEMPERATURE),
-                    user=session.conversation_id,
-                )
+                if model.startswith(("o1-mini", "o1-preview")):
+                    # These models don't support tools and reasoning effort
```

Comment:
What is different in these args? I can't tell.

Perhaps you can make a dict of args liek `model_args` so its easier to see the differences between versions, then expand the keyword args with `**model_args`

Reply:
I think it's easier to use the `**` operator to pass a dict as keyword args to the function. That way we can define the generic ones, and then add the other args conditionally.

Reply:
```python
kwargs = {
  "model": model,
  "prompt": prompt,
}
if model.startswith("o"):
  kwargs["reasoning"] = ...

result = await client.chat.completions.create(**kwargs)
```

Reply:
Done, looks much better now!